java/sage/Carny.java

```diff
@@ -1159,61 +1617,87 @@ private void stdProcessing()
     newWPMap.keySet().removeAll(watchedPotsToClear);
 
     if (Sage.DBG) System.out.println("CARNY Negative Energy Size: " + (blackBalled.size() + stoners.length));
-    for (Airing badAir : blackBalled)
+    for (int i = 0, blackBalledSize = blackBalled.size(); i < blackBalledSize; i++)
     {
+      Airing badAir = blackBalled.get(i);
```

Comment:
That's not a bad point there. I guess I didn't look at that piece very hard because it finishes in less than 200ms even with your database.